core/src/main/java/org/jdbi/v3/core/argument/PrimitivesArgumentFactory.java

```diff
@@ -13,10 +13,27 @@
  */
 package org.jdbi.v3.core.argument;
 
+import java.lang.reflect.Type;
 import java.sql.PreparedStatement;
 import java.sql.Types;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.IdentityHashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.Function;
 
+import org.jdbi.v3.core.argument.internal.StatementBinder;
+import org.jdbi.v3.core.argument.internal.strategies.LoggableBinderArgument;
+import org.jdbi.v3.core.config.ConfigRegistry;
+
+import static org.jdbi.v3.core.generic.GenericTypes.getErasedType;
+
+// TODO this factory isn't actually preparable anymore
 class PrimitivesArgumentFactory extends DelegatingArgumentFactory {
+    private final Map<Class<?>, BiFunction<Object, ConfigRegistry, Argument>> builders = new IdentityHashMap<>();
```

Comment:
It's weird to extend `DelegatingArgumentFactory` but not use its functionality, but instead replicate it here.  Feels like a design smell.

Reply:
Removing the `extends` would technically be breaking API, though for this case I'm hoping you'll say we can shrug that off